Mutagen.Bethesda.Analyzers.Skyrim/Record/Dialog/Responses/LinksToDifferentSpeakerAnalyzer.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Analyzers.Skyrim.Util;
+using Mutagen.Bethesda.Skyrim;
+using Noggog;
+
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Record.Dialog.Responses;
+
+public class LinksToDifferentSpeakerAnalyzer : IContextualRecordAnalyzer<IDialogTopicGetter>
+{
+    public static readonly TopicDefinition<IDialogTopicGetter> LinksToDifferentSpeaker = MutagenTopicBuilder.FromDiscussion(
+            468,
+            "Links to Different Speaker",
+            Severity.Error)
+        .WithFormatting<IDialogTopicGetter>("Topic has response that links to topic {0} that has no speakers in common with the responses linking to it");
+
+    public IEnumerable<TopicDefinition> Topics { get; } = [LinksToDifferentSpeaker];
+
+    public void AnalyzeRecord(ContextualRecordAnalyzerParams<IDialogTopicGetter> param)
+    {
+        var dialogTopic = param.Record;
+
+        var voiceTypeAssetLookup = VoiceTypeAssetLookupUtil.GetVoiceTypeAssetLookup(param.LinkCache);
```

Comment:
I wonder if we can add some "downstream" cache concept.   For ex, the params could provide something like
`param.RetrieveCache<VoiceTypeAssetLookup>()`
where `VoiceTypeAssetLookup` implements an interface like
```
public interface IDownstreamCache
{
     void Prep(ILinkCache linkCache, IAssetLinkCache assetCache);
}
```

The params object could then be the one to do/handle most of the work in VoiceTypeAssetLookupUtil.  Additionally, the engine would be better able to invalidate it all when the linkcache changed

Reply:
It would be a feature of the analyzer engine, and exposed via the parameter objects.  This would be similar to a container-like api, where you might do `param.ResolveCache<VoiceTypeAssetLookup>()`.   The analyzer would get one back without knowing how it was made, or knowing whether it was the same one as the last time it ran.

This would let the engine:
- Handle its construction.  We might make a component that `ICanMake<VoiceTypeAssetLookup>` that might itself `ResolveCache<IAssetLinkCache>()`, for ex
- Handle when it should be invalidated.  Engine could track when a new linkcache is required, and thus which downstream caches to purge so new ones are made